### PR TITLE
cam6_3_081: Misc tag

### LINKED ISSUE #644: The interpolate_output namelist flag results in bad output values
### What happened?

As reported by @brianpm , a recent CAM simulation with the `ne30pg3_ne30pg3_mg17` and `ne30_g17` grids and the `interpolate_output` namelist flag(s) set to `.true.` resulted in history files with bad/missing values (history variables had O(10^-38) values).  

It should be noted that the non-interpolated history output was fine, so this is likely an issue with the interpolation itself.

### What are the steps to reproduce the bug?

Not entirely sure yet, but based on the example case below it appears that running an F2000climo compset for two days with one of the grids above and the following `user_nl_cam` settings:

```
nhtfrq = 0,-1,-1
mfilt = 1,24,24
ndens = 2,2,2
empty_htapes=.false.
interpolate_output=.true.,.true.,.true.
interpolate_nlat=192,192,192
interpolate_nlon=288,288,288

FINCL2 = 'U','V','T','Q','SOLIN','FSNT','FLUT','PRECC','PRECT'
FINCL3 = 'U','V','T','Q','SOLIN','FSNT','FLUT','PRECC','PRECT'

iradsw = -1
iradlw = -1
use_rad_dt_cosz = .false.
spectralflux = .false.
```

results in the bad history file output.  I will add a comment to this issue if I can find an even simpler setup which produces the same error.

### What CAM tag were you using?

Not sure, but will try with the latest tag and report the results here.

### What machine were you running CAM on?

CISL machine (e.g. cheyenne)

### What compiler were you using?

Intel

### Path to a case directory, if applicable

/glade/work/brianpm/my_cases/test_cases/vanilla.f2000climo.ne30pg3.testinterp02

### Will you be addressing this bug yourself?

Yes

### Extra info

_No response_

### LINKED ISSUE #667: More restart non-bit-for-bit occurrences
### What happened?

From Keith Lindsay via email:

I have confirmed that my clone of one of Cecile's fully coupled experiments is not restarting exactly, when attempting to restart on particular dates.

The experiment restarts exactly early in the run, such as 0001-01-02 and 0001-01-03.
It also restarts exactly on 0001-12-01.
However, it does not restart exactly on 0001-12-31 or 0002-01-01.

In my tests of exact restartability, I've enabled cpl hist files (approximately) every timestep. (I need to move where history writing in the nuopc run sequence occurs to get this to happen.) So I'm getting a record of fields getting passed to and from the coupler. The first divergence of fields after a restart is in some fields sent from CAM to the coupler. Here's the RMS lines from cprnc comparing the cpl hist files:
 RMS atmImp_Faxa_rainc                7.1978E-15            NORMALIZED  4.7962E-10
 RMS atmImp_Faxa_snowc                1.5895E-16            NORMALIZED  3.5183E-09
 RMS atmImp_Sa_dens                   1.8034E-13            NORMALIZED  1.5064E-13
 RMS atmImp_Sa_pbot                   1.0914E-08            NORMALIZED  1.1155E-13
 RMS atmImp_Sa_ptem                   1.8113E-11            NORMALIZED  6.3387E-14
 RMS atmImp_Sa_shum                   1.2418E-14            NORMALIZED  1.3697E-12
 RMS atmImp_Sa_tbot                   1.8035E-11            NORMALIZED  6.3250E-14
 RMS atmImp_Sa_u                      7.9879E-13            NORMALIZED  1.5167E-13
 RMS atmImp_Sa_v                      4.5647E-12            NORMALIZED  9.7223E-13
 RMS atmImp_Sa_z                      9.1689E-12            NORMALIZED  1.4462E-13

So the differences are pretty small. But they grow.

This experiment has a number of CAM forcings with type CYCLICAL. Given that the problem occurs at the end of the year, but not only on Jan 1, my initial hypothesis is that there is a roundoff difference after restart in one of these forcing fields. That is, how the interpolation weights in time across the year boundary are computed.

I don't have high frequency CAM output enabled, yet, so I'm currently only able to diagnose differences in the fields the CAM sends to the coupler.

Given what I'm seeing, I'm guessing that this is reproducible in an F compset, and probably at lower resolution.

Do you have a preference on how I, or we, might proceed on further investigating this?

### What are the steps to reproduce the bug?

See above description

### What CAM tag were you using?

branch which Cecile is using

### What machine were you running CAM on?

CISL machine (e.g. cheyenne)

### What compiler were you using?

Intel

### Path to a case directory, if applicable

_No response_

### Will you be addressing this bug yourself?

No

### Extra info

This issue will be evolving and becoming more specific as we gain more information

### LINKED ISSUE #674: convect_shallow.F90 has CIN in an addfld call, but there is no corresponding outfld call
### What happened?

A user in the forum reported running with an FHIST compset, and added CIN to his fincl call.  His output was filled with zeroes.  A visual inspection of physics/cam/convect_shallow.F90 finds an addfld call for this variable, but an outfld call does not exist in this routine.  I was also unable to find an outfld call for CIN using grep through the source code tree.

### What are the steps to reproduce the bug?

I did not run a job to verify this

### What CAM tag were you using?

cam6_3_072

### What machine were you running CAM on?

Other (please explain below)

### What compiler were you using?

Other (please specify below)

### Path to a case directory, if applicable

_No response_

### Will you be addressing this bug yourself?

No

### Extra info

This is being reported based on a forum user's report and a visual inspection of the code.

### LINKED ISSUE #679: indexing bug for optional argument PS for get_dp subroutine of cam_thermo.F90
### What happened?

The optional output argument PS is passed into get_dp as an (i,j) array but used within the procedures as (i,k).  An out of bounds reference on the array causes the model to halt when run under DEBUG with NAG.
PS should be passed in as (:,j) and worked with inside the get_dp routines as a single rank variable dimensioned by (i)

A diff of cam_thermo without/with fix:

```
diff ../../collections/cam6_3_078.FCSCAM/src/utils/  ../../collections/cam6_3_080_ebudget_dev_update.102022/src/utils/cam_thermo.F90
_600c600
<      real(r8), optional,intent(out) :: ps(:, :)                  ! surface pressure (if ps present then ptop'
---
>      real(r8), optional,intent(out) :: ps(:)                     ! surface pressure (if ps present then ptop
635c635
<              ps(idx, kdx) = ps(idx, kdx) + dp(idx, kdx)
---
>              ps(idx) = ps(idx) + dp(idx, kdx)
658,659c658,664
<        call get_dp(tracer(:, jdx, :, :), mixing_ratio, active_species_idx,  &
<                dp_dry(:, jdx, :), dp(:, jdx, :), ps=ps, ptop=ptop)
---
>        if (present(ps).and.present(ptop)) then
>           call get_dp(tracer(:, jdx, :, :), mixing_ratio, active_species_idx,  &
>                dp_dry(:, jdx, :), dp(:, jdx, :), ps=ps(:,jdx), ptop=ptop)
>        else
>           call get_dp(tracer(:, jdx, :, :), mixing_ratio, active_species_idx,  &
>                dp_dry(:, jdx, :), dp(:, jdx, :))
>        end if
```

### What are the steps to reproduce the bug?

We are calling get_dp in some new budgeting code running with DEBUG under NAG.  Any call to get_dp where the optional argument PS is specified should reproduce the error using DEBUG under NAG.

### What CAM tag were you using?

cam6_3_080

### What machine were you running CAM on?

CGD machine (e.g. izumi)

### What compiler were you using?

NAG

### Path to a case directory, if applicable

_No response_

### Will you be addressing this bug yourself?

No

### Extra info

_No response_

### PR DESCRIPTION
**Pull request by peverwhee**
The only updates that haven't already been reviewed are the CAM thermo bug and addfld call removal (listed first)

### CAM thermo bug
Indexing bugfix for get_dp in cam_thermo.F90
closes #679 

### Remove CIN addfld call from convect_shallow
Removing because it doesn't have an associated outfld call.
closes #674

### Interpolate Output Fixes
Allows for single-precision (e.g. ndens = 2) interpolated output, and allows for variables with missing values (e.g. AODVIS) to be output correctly in the standard monthly average (h0) files, at least for certain SE-CSLAM grids.
closes #658
closes #644 
addresses #564 
was reviewed in #659 

### DAE Test Fixes
Needs a cmeps externals update to fix tests fully. Will bring cmeps external in with future Externals update PR.
closes #671 
addresses #670 
was reviewed in #671

### Data reader bug in WACCMX
fixes restart issue Cecile encountered
closes #672 
closes #667 
was reviewed in #672

### CODE REVIEW

File: doc/ChangeLog [hunk 1]
```diff
@@ -1,5 +1,144 @@
...
+List all files eliminated: N/A
+
+List all files added and what they do:
+A       cime_config/testdefs/testmods_dirs/cam/outfrq1d_14dec/shell_commands
+A	cime_config/testdefs/testmods_dirs/cam/outfrq1d_14dec/user_nl_cam
+A	cime_config/testdefs/testmods_dirs/cam/outfrq1d_14dec/user_nl_clm
+A	cime_config/testdefs/testmods_dirs/cam/outfrq1d_14dec_L32wsc/shell_commands
+A	cime_config/testdefs/testmods_dirs/cam/outfrq1d_14dec_L32wsc/user_nl_cam
+A	cime_config/testdefs/testmods_dirs/cam/outfrq1d_14dec_L32wsc/user_nl_clm
+           - add testmods for new WACCMX restart tests
```
Comment (fvitt) [inline doc/ChangeLog:38]:
```suggestion
           - add testmods for WACCM-SC chemistry restarts across the Dec 16 boundary
```

File: doc/ChangeLog [hunk 2]
```diff
@@ -1,5 +1,144 @@
...
+
+List all existing files that have been modified, and describe the changes:
+M       cime_config/testdefs/testlist_cam.xml
+           - Add new WACCMX restart tests
```
Comment (fvitt) [inline doc/ChangeLog:42]:
```suggestion
           - Add new tests for WACCM-SC chemistry restarts across the Dec 16 boundary
```

File: doc/ChangeLog [hunk 3]
```diff
@@ -1,5 +1,144 @@
...
+M       src/control/cam_history
+M	src/dynamics/se/dycore/interpolate_mod.F90
+M	src/dynamics/se/interp_mod.F90
+           - interpolate_output fixes
+
+M       src/cpl/nuopc/atm_comp_nuopc.F90
+           - DAE test fixes
+
+M	src/physics/cam/boundarydata.F90
+           - WACCMX fix bug data reader
```
Comment (fvitt) [inline doc/ChangeLog:53]:
```suggestion
           - fix bug in data reader used for WACCM-SC loss rates input
```

File: doc/ChangeLog [hunk 4]
```diff
@@ -1,5 +1,144 @@
...
+
+cheyenne/intel/aux_cam: 
+
+izumi/nag/aux_cam: 
+
+izumi/gnu/aux_cam: 
+
+===============================================================
+
+Tag name: cam6_3_080
```
Comment (fvitt) [inline doc/ChangeLog:74]: Looks like you have inadvertently repeated the change log for tag `cam6_3_080`

**Review by fvitt — CHANGES_REQUESTED** [review]
some initial suggestions

File: doc/ChangeLog (hunk 4 above)
Comment (peverwhee) [reply doc/ChangeLog:74]: indeed I did! removed.

File: doc/ChangeLog (hunk 3 above)
Comment (peverwhee) [reply doc/ChangeLog:53]: changed.

File: doc/ChangeLog (hunk 2 above)
Comment (peverwhee) [reply doc/ChangeLog:42]: changed.

File: doc/ChangeLog (hunk 1 above)
Comment (peverwhee) [reply doc/ChangeLog:38]: thanks, francis! changed.

**Review by fvitt — APPROVED** [review]
Seem okay to me. Thanks.

File: doc/ChangeLog [hunk 5]
```diff
@@ -1,5 +1,78 @@
 ===============================================================
 
+Tag name: cam6_3_081
+Originator(s): jtruesdal, nusbaume, mvertens, fvitt, peverwhee
+Date: October 26, 2022
+One-line Summary: Misc tag
+Github PR URL:
```
Comment (nusbaume) [inline doc/ChangeLog:7]: Just a reminder to not forget the PR URL here:

`https://github.com/ESCOMP/CAM/pull/680`

**Review by nusbaume — APPROVED** [review]
Looks great, thanks @peverwhee!

**Comment by peverwhee** [comment]
@fvitt I'm getting DIFFs for this test: ERP_Ln9_Vnuopc.f19_f19_mg17.FWsc1850.cheyenne_intel.cam-outfrq9s

When you get a spare moment, can you take a look at these differences? Is this expected?

Cheyenne test directory: /glade/scratch/courtneyp/aux_cam_20221026152937/ERP_Ln9_Vnuopc.f19_f19_mg17.FWsc1850.cheyenne_intel.cam-outfrq9s.GC.aux_cam_20221026152937

**Comment by fvitt** [comment]
@peverwhee Differences are expected for WACCM-SC due to the bug fix boundarydata.F90.  There is a correction in the initial read of multiple fields from an input file.

**Comment by peverwhee** [comment]
@fvitt Thanks Francis

**Review by jtruesdal — APPROVED** [review]
Good job.  Reviewed this for Cheryl a little while ago with comments so everything looks good to go.

File: doc/ChangeLog (hunk 5 above)
Comment (peverwhee) [reply doc/ChangeLog:7]: added
